crates/bevy_render/src/shader/shader.rs

```diff
@@ -294,6 +295,51 @@ impl ShaderStages {
     }
 }
 
+/// Compute stages in a shader program
+#[derive(Clone, Debug)]
+pub struct ComputeShaderStages {
+    pub compute: Handle<Shader>,
+}
+
+pub struct ComputeShaderStagesIterator<'a> {
+    shader_stages: &'a ComputeShaderStages,
+    state: u32,
+}
+
+impl<'a> Iterator for ComputeShaderStagesIterator<'a> {
+    type Item = Handle<Shader>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let ret = match self.state {
+            0 => Some(self.shader_stages.compute.clone_weak()),
+            _ => None,
+        };
+        self.state += 1;
+        ret
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (1, Some(1))
+    }
+}
```

Comment:
We could simplify this to use an `Option<_>`

```rust
pub struct ComputerShader... {
    shared_stages: Option<&'a ComputeShaderStages>,
}

impl<'a> Iterator for ... {
    fn next(&mut self) -> Option<Self::Item> {
        self.take().map(|stages| stages.compute.clone_weak())
    }
}
```

Reply:
A lot of this was pulled directly from the other shader stuff. I agree it probably could cleaned up though. I'll see what I can do.